File: franc.go
package money

type franc struct {
	value
}

func NewFranc(amount int) *franc {
	franc := new(franc)
	franc.amount = amount
	return franc
}

func (f *franc) times(multiplier int) value {
	return NewFranc(f.amount * multiplier).value
}

File: src/Money/Franc.php
<?php

namespace Money;

class Franc
{
    private $Amount;

    public function __construct(int $amount)
    {
        $this->Amount = $amount;
    }

    public function times(int $multiplier)
    {
        return new Franc($this->Amount * $multiplier);
    }

    public function equals(Franc $franc): bool
    {
        return $this->Amount == $franc->Amount;
    }
}
